pkg/test/framework/components/istio/operator.go

```diff
@@ -286,19 +294,59 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	// Wait for all of the control planes to be started.
 	for _, cluster := range env.KubeClusters {
 		// TODO in centralIstiod case, webhook is only installed in control plane cluster
-		if !env.IsControlPlaneCluster(cluster) && isCentralIstio(env, cfg) {
+		if !env.IsControlPlaneCluster(cluster) && isCentralIstio(env, cfg) || cfg.UsingRemoteKubeConfig {
 			continue
 		}
 		if err := waitForControlPlane(ctx, i, cluster, cfg); err != nil {
 			return nil, err
 		}
 	}
 
+	// started to create related endpoints for cases istiod is using remote KUBECONFIG
+	if cfg.UsingRemoteKubeConfig {
```

Comment:
It feels like most of this function (`deploy`) now has conditionals based on `UsingRemoteKubeConfig`, making it hard to follow... would it be possible to refactor in a way that makes this cleaner?

Reply:
Make sense. I  made all this small check into one big one and move logci of UsingRemoteKubeConfig  at the end.

Reply:
Cool, it seems more readable now. Given the length of the function and the extra indentation, I wonder if it would be helpful to move the two branches into helper functions

Reply:
Rather than adding another branch in this one function, can we just 

```
if UsingRemoteKubeConfig {
  deployWithRemoteKubeConfig()
}
```

Reply:
Actually yeah, what stevenctl said seems preferable, leaving most of deploy() untouched

Reply:
fixed